docs/conf.py

```diff
@@ -57,9 +61,27 @@
     "sphinx_issues",
 ]
 
+
+shutil.copy("../src/core/pyproxy.ts", "../src/js/pyproxy.gen.ts")
+shutil.copy("../src/core/error_handling.ts", "../src/js/error_handling.gen.ts")
+
+
+def remove_pyproxy_gen_ts():
+    Path("../src/js/pyproxy.gen.ts").unlink(missing_ok=True)
+
+
+atexit.register(remove_pyproxy_gen_ts)
+
+os.environ["PATH"] += f':{str(Path("../src/js/node_modules/.bin").resolve())}'
+
+if not shutil.which("typescipt") or not shutil.which("typedoc"):
+    subprocess.run(["npm", "ci"], cwd="../src/js")
```

Comment:
We might want to wrap this into some `if` condition checking that we are indeed running in sphinx, and not say that this file was mistakenly imported during pytest discovery.

Reply:
Is there a good way to detect being inside readthedocs? We probably want to fail rather than run `npm ci` if the user is running this. I'd prefer to run it in a separate script but readthedocs doesn't seem to support that. Another option might be to move some stuff into a separate `conf_readthedocs.py` which could read as follows:
```py
import subprocess
subprocess.run(["npm", "ci"], cwd="../src/js")
from conf import *
```

Reply:
Someone here has a technique to detect sphinx:
https://github.com/sphinx-doc/sphinx/issues/9805

Reply:
For being inside readthedocs see https://github.com/pyodide/pyodide/blob/main/docs/conf.py#L130

Well contributors still might want to build the docs locally, and ideally, it shouldn't fail even if it doesn't build everything